Homework #11 done

Vozzhd opened the pull request.

Vozzhd committed:
Sprint №11 HomeWork done, need refactor

Added:
-placeholders
-API call
-dependencies
Changed:
-recyclerView

Vozzhd committed:
Sprint №11 HomeWork done
-check null cast from api data
-ArrayList changed to MutableList
-changed dimens
-changed XML's

taikonavt commented on app/src/main/res/layout/activity_search.xml:
```diff
@@ -42,8 +46,44 @@
+
+    <com.google.android.material.button.MaterialButton
+        android:id="@+id/refreshButton"
+        style="@style/RegularTextView"
```
ты здесь применяешь стиль, а потом переопределяешь почти все параметры этого стиля. Стиль хорош тем, что если, например, дизайнер решит поменять цвет текста на кнопках во всем приложении и у тебя ко всем нужным кнопкам применен один стиль, то ты поменяешь цвет только в стиле и он применится ко всем кнопкам. А если у тебя кнопки сделаны без использования стиля, то придется менять цвет в каждой кнопке.

Vozzhd replied:
Решено, выделил в отдельный стиль

taikonavt commented on app/src/main/res/layout/activity_search.xml:
```diff
@@ -42,8 +46,44 @@
+        android:layout_marginTop="458dp"
+        android:backgroundTint="?attr/colorOnPrimary"
+        android:gravity="center"
+        android:text="Обновить"
```
текст нужно вынести в strings

Vozzhd replied:
Сделано

taikonavt commented on app/src/main/res/layout/activity_search.xml:
```diff
@@ -42,8 +46,44 @@
         android:backgroundTint="@color/transparent"
         app:layoutManager="androidx.recyclerview.widget.LinearLayoutManager" />
+
+    <com.google.android.material.button.MaterialButton
```
кнопка выглядит не так, как на макете

Vozzhd replied:
Исправил

taikonavt commented on app/src/main/res/layout/activity_search.xml:
```diff
@@ -42,8 +46,44 @@
+        android:gravity="center"
+        android:text="Обновить"
+        android:textColor="?attr/colorPrimary"
+        android:textSize="11dp"
```
размер текста указываем в sp

Vozzhd replied:
Исправлено

taikonavt commented on app/src/main/java/com/practicum/playlistmaker/SearchActivity.kt:
```diff
@@ -43,6 +65,56 @@ class SearchActivity : AppCompatActivity() {
+                                trackList.addAll(checkedResponseBody)
+                                trackViewAdapter.notifyDataSetChanged()  //Надо бы поменять
+                            }
+                            if (trackList.isEmpty()) {
```
если получить ошибку "проблема со связью", а потом нажать на "обновить" и получить пустой список, то кнопка "обновить" не пропадает. На макете ее нет

Vozzhd replied:
Исправил, немного поменял подход к экрану.

taikonavt commented on app/src/main/java/com/practicum/playlistmaker/SearchActivity.kt:
```diff
@@ -5,13 +5,32 @@ import android.os.Bundle
+        .build()
+
+    private val itunesService = retrofit.create(ItunesAPI::class.java)
+    private val trackList = mutableListOf<Track>()
```
думаю, нет смысла создавать дополнительный trackList в активити. У тебя уже есть trackList для хранения треков в адаптере, получается что это его дубль. Можно обновлять сразу список из адаптера. Как вариант можно в адаптере создать метод fun updateList(newList: ArrayList){ trackList.clear() trackList.addAll(newList) notifyDataSetChanged() } Будет меньше дублируемого кода

Vozzhd replied:
Задумку понял, идею понял, пока не реализовал как проверять его на содержание. В будущем исправлю.

taikonavt commented on app/src/main/res/layout/activity_search.xml:
```diff
@@ -17,15 +17,19 @@
         android:layout_marginTop="@dimen/marginFromTopForSearchField">
 
         <EditText
             android:id="@+id/findField"
```
цвет введенного текста для светлой темы в edittext серый. На макете он черный

Vozzhd replied:
![image](https://github.com/Vozzhd/PlaylistMaker/assets/144613563/1a9620eb-61a4-45f8-bdcd-396ebad1e8fd)
Он меняется он макета к макету)) я брал цвета из общего макета приложения.
UPD: А, всё, туплю.  Цвет **уже введённого** действительно чёрный. Исправил, пока локально.

Vozzhd committed:
Sprint №11 HomeWork after review
Add:
-PlaceholderState enum
-FrameLayout for PlaceholderState
Changed:
-placeholderError Logic
-styles
-activity_search.xml

taikonavt commented on app/src/main/java/com/practicum/playlistmaker/SearchActivity.kt:
```diff
@@ -67,51 +70,68 @@ class SearchActivity : AppCompatActivity() {
+            }
+        }
+
+        fun setPlaceholderState(placeholderState: PlaceholderState) {
+            when (placeholderState) {
+                PlaceholderState.GOOD -> {
+                    placeholderErrorLayout.visibility = View.GONE
+                }
+                PlaceholderState.BAD_REQUEST -> {
+                    showMessage(getString(R.string.nothing_found))
+                    errorImage.setImageResource(R.drawable.search_error)
+                    refreshButton.visibility = View.GONE
+                    placeholderErrorLayout.visibility = View.VISIBLE
+                }
+                PlaceholderState.NO_CONNECTION -> {
+                    showMessage(getString(R.string.something_went_wrong))
+                    errorImage.setImageResource(R.drawable.connection_error)
+                    refreshButton.visibility = View.VISIBLE
+                    placeholderErrorLayout.visibility = View.VISIBLE
+                }
```
хороший подход
